src/realm/object-store/shared_realm.cpp

```diff
@@ -254,7 +254,11 @@ void Realm::read_schema_from_group_if_needed()
         }
     }
     else {
-        ObjectStore::verify_valid_external_changes(m_schema.compare(schema, m_config.schema_mode));
+        if (!m_config.is_schema_additive()) {
+            // if the schema is additive the 2 schemas will be merged, so there is nothing to check. All the tables
+            // will still be available
+            ObjectStore::verify_valid_external_changes(m_schema.compare(schema, m_config.schema_mode));
+        }
         m_schema.copy_keys_from(schema);
```

Comment:
It is not clear to me if I need to consider the schema mode at this stage, it seems to me that I should probably do. @tgoyne